aixtend/factories/metric_factory.py

```diff
@@ -0,0 +1,76 @@
+__author__='shreyassharma'
+
+"""
+Copyright 2022 The aiXplain SDK authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Author: Duraikrishna Selvaraju, Thiago Castro Ferreira, Shreyas Sharma and Lucas Pavanelli
+Date: December 1st 2022
+Description:
+    Metric Factory Class
+"""
+
+import logging
+from typing import List
+from aixtend.modules.metric import Metric
+from aixtend.utils import config
+from aixtend.utils.file_utils import _request_with_retry
+
+class MetricFactory:
+    def __init__(self) -> None:
+        self.api_key = config.TEAM_API_KEY
+        self.backend_url = config.BENCHMARKS_BACKEND_URL
+    
+
+    @staticmethod
+    def _create_metric_from_response(self, response: dict) -> Metric:
+        """Coverts response Json to 'Metric' object
+
+        Args:
+            response (dict): Json from API
+
+        Returns:
+            Metric: Coverted 'Metric' object
+        """
+        return Metric(response['id'], response['name'], response['description'])
+    
+
+
+
+
+    def list_metrics(self, task:str) -> List[Metric]:
+        """Get list of supported metrics for a given task
+
+        Args:
+            task (str): Task to get metric for
+
+        Returns:
+            List[Metric]: List of supported metrics
+        """
+        try:
+            url = f"{self.backend_url}/sdk/scores?function={task}"
+            headers = {
+            'Authorization': f"Token {self.api_key}",
+            'Content-Type': 'application/json'
+            }
+            r = _request_with_retry("get", url, headers=headers)
+            resp = r.json()
```

Comment:
My suggestion is that there should be a class called BE API which has a method that gets or post or put and takes a path as a parameter something like BE_API.post("scores?function={task}") and internally it will do the call and pick up the token and URL from the env

Reply:
@shreyasXplain this comment is unresolved.

Reply:
@anasbakro This would be okay for the team-api key but for model api-key we would still need to pass the API key(for model/pipeline) as a parameter.  Also we would need to still have this function to send normal requests for tasks like downloading files etc.
Nonetheless we can still in addition to current function develop the class. What do you think?